calico/felix/endpoint.py

```diff
@@ -130,6 +132,9 @@ def on_endpoint_update(self, endpoint_id, endpoint):
             self.endpoint_id_by_iface_name[endpoint["name"]] = endpoint_id
 
         if endpoint and endpoint["host"] == self.config.HOSTNAME:
+        # MD4: Do we really need to do all the above work for non-local
+        # endpoints?  Since the config for hostname is always available, can't
+        # we move this check to the top?
```

Comment:
Actually, I'm wrong on that.  We don't need to store non-local endpoints any more.  We do need to do that in the tag management code but I can't see any reason to do it here.